academic-services/knowledge-exploration-service/concepts-queries.md

```diff
@@ -0,0 +1,113 @@
+
+## Stage 2: Generate valid interpretation hypotheses
+
+:::image type="content" source="media/concept-semantic-interpretation-hypothesis.png" alt-text="MAKES Semantic Interpretation Generation Architecture - Generate interpretation hypothesis":::
+
+Semantic interpretation generation is broken into two important stages. The first of these is generating interpretation hypotheses using a [Context Sensitive Grammar (CSG)](https://en.wikipedia.org/wiki/Context-sensitive_grammar) conforming to the [Speech Recognition Grammar Specification (SRGS)](https://www.w3.org/TR/speech-grammar/).
+
+Generally SRGS grammars are context-free, meaning grammar rules can be matched lacking any additional context of previously matched rules. MAKES changes this behavior by including a *grammar context* that enables rules to reference metadata from previously matched rules.
+
+The default SRGS grammar used by MAKES is very simple, with a single top-level rule which is allowed to repeat until all query terms have been processed.
+
+* GetPapers
+  * For current term(s) do *one* of the following:
+    * Match term to paper title words available in the current grammar context
+    * Match terms to paper authors available in the current grammar context
+    * ...
+  * If match found, add new match to the grammar context, then either move to the next query segment and repeat top level rule or if at end of query move to resolve the hypothesis
+  * If no match found, terminate current grammar context
+
+It's important to understand that each hypothesis generates *sub-hypotheses* in the form of an expansion-tree, and each tree branch *gets its own unique copy of the grammar context*. This allows MAKES to generate many potential hypothesis that fully explore the search space for a given query.
+
+## Stage 3: Resolve valid interpretation hypothesis
+
+:::image type="content" source="media/concept-semantic-interpretation-resolve.png" alt-text="MAKES Semantic Interpretation Generation Architecture - Resolve valid hypothesis":::
+
+Once a valid hypothesis has been found, MAKES *resolves* the hypothesis into an interpretation by generating and storing the following:
+
+* Probability score for the hypothesis, which is generally the static rank of the top-most entity returned by the hypothesis plus any alternative [weight penalties imposed by the SRGS grammar](https://www.w3.org/TR/speech-grammar/#S2.4)
+* XML parse which includes the grammar rules and attributes matched to generate the hypothesis
+* [Structured query expression](concepts-query-expressions.md) that can be used to find all entity results matching the hypothesis
+* Optionally the top matching entities themselves can be also be returned
+
+## Stage 4: Generate interpretation response
+
+The final stage of semantic interpretation generation is aggregating all found valid hypotheses and generating a final JSON response.
```

Comment:
The final stage of semantic interpretation generation aggregates all the hypotheses that were found and generates a JSON response.